appnav/src/main/kotlin/io/element/android/appnav/LoggedInFlowNode.kt

```diff
@@ -127,8 +131,18 @@ class LoggedInFlowNode @AssistedInject constructor(
     )
 
     private val verificationListener = object : SessionVerificationServiceListener {
-        override fun onIncomingSessionRequest(sessionVerificationRequestDetails: SessionVerificationRequestDetails) {
-            backstack.singleTop(NavTarget.IncomingVerificationRequest(sessionVerificationRequestDetails))
+        override fun onIncomingSessionRequest(verificationRequest: VerificationRequest.Incoming) {
+            // Without this launch the rendering and actual state of this Appyx node's children gets out of sync, resulting in a crash.
+            // This might be because this method is called back from Rust in a background thread.
+            MainScope().launch {
```

Comment:
Can `RustSessionVerificationService` ensure that `onIncomingSessionRequest` is invoked on the same thread that was used when registering the listener?

Reply:
I don't think so, this is due to how JNA works, it'll create a separate thread for this callback and use it when receiving data from Rust. Maybe we could somehow retain the thread instance and force the callback to somehow run some runnable in it, but it seems like an overkill.

Reply:
> retain the thread instance and force the callback to somehow run some runnable in it

This is what I had in mind, but OK to keep it like this :)